deepchecks/base/dataset.py

```diff
@@ -122,28 +117,41 @@ def __init__(
         self._data = df.copy()
 
         # Validations
-        if label is not None:
-            if label.shape[0] != self._data.shape[0]:
-                raise DeepchecksValueError('Number of samples of label and data must be equal')
-            if len(label.shape) > 1 and label.shape[1] != 1:
-                raise DeepchecksValueError('Label must be a column vector')
-            # Make tests to prevent overriding user column
-            if label_name is None:
-                label_name = 'target'
-                if label_name in self._data.columns:
-                    raise DeepchecksValueError(f'Data has column with name "{label_name}", use label_name parameter'
-                                               ' to set column name for label which does\'t exists in the data')
-            else:
-                if label_name in self._data.columns:
-                    raise DeepchecksValueError('Can\'t pass label with label_name that exists in the data. change '
-                                               'the label_name parameter')
-
-            # If passed label is a pandas object, check that indexes match, else set column as is with provided values
-            if isinstance(label, (pd.Series, pd.DataFrame)):
-                pd.testing.assert_index_equal(self._data.index, label.index)
-                self._data[label_name] = label
+        if label is None:
+            label_name = None
+        else:
+            if isinstance(label, (pd.Series, pd.DataFrame, np.ndarray)):
+                # Validate label shape
+                if label.shape[0] != self._data.shape[0]:
+                    raise DeepchecksValueError('Number of samples of label and data must be equal')
+                if len(label.shape) > 1 and label.shape[1] != 1:
```

Comment:
doesn't cover cases like if someone transfers a 3 (or more) dimensional array

Reply:
do we support these use cases?

Reply:
doesn't seem like something we should support...
in other words - exactly my question (that we should verify that the user passed valid input)
that here someone can pass a 3d np.ndarray, e.g. of shape (<num_samples_in_data>, 1, 42) , then the code will run successfully....

(That's also what I meant in the tests - verifying that it fails as it should if someone sends a DataFrame with more than one column, and same for ndarrays with too many dimensions)

btw coming to thing of it - for numpy I would sugggest separate treatment - using np.squeeze, and then verifying that it's a 1d array with the correct length (cause np arrays have various weird dimension behaviors.... and this also makes us note care if it's a "column vector" or a "row vector")

Reply:
we looked at pandas, and they are not allowing 3+ dimensions. so right now I'll check for a maximum of 2-dim, and that at least one of them is equal 1 (meaning we accept either a column or row vector). we can support any 2-dim shape with squeeze but it might be confusing

Reply:
I was talking about numpy for the 3d (plus)
pandas indeed has only up to 2-dim
but numpy has unlimited (that's why offered the squeeze)

Reply:
yea I meant we looked at how pandas handled numpy arrays as input

Reply:
Yeah pandas doesn't allow more then 2-d and I agree that there is no reason to allow more. The user can squeeze by himself IMO

Reply:
oh, ok, cool :)

Reply:
oh, cool :)